src/avram/database.cr

```diff
@@ -1,38 +1,65 @@
-class Avram::Repo
+abstract class Avram::Database
   alias FiberId = UInt64
 
   @@db : DB::Database? = nil
-  private class_getter transactions = {} of FiberId => DB::Transaction
+  class_getter transactions = {} of FiberId => DB::Transaction
 
-  Habitat.create do
-    setting url : String
-    setting lazy_load_enabled : Bool = true
-    setting logger : Dexter::Logger = Dexter::Logger.new(nil)
+  macro inherited
+    Habitat.create do
+      setting url : String
+    end
+  end
+
+  # TODO: Add a fallback 'configure' that raises at compile time and says
+  # to create a Database class
+
+  # Rollback the current transaction
+  def self.rollback
+    new.rollback
+  end
+
+  # runs a SQL `TRUNCATE` on all tables in the database
+  def self.truncate
+    new.truncate
+  end
+
+  def self.transaction
```

Comment:
Could you add a quick comment on what this does? Will make it easier to add to guides later.

Reply:
Yup!